src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,152 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private static User bob;
+    private static User alice;
+    private RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+    private StorageDaoImpl storageDao = new StorageDaoImpl();
+
+    @BeforeAll
+    static void beforeAll() {
+        bob = new User();
+        bob.setLogin("bob");
+        bob.setPassword("1234567");
+        bob.setAge(21);
+        alice = new User();
+        alice.setLogin("alice");
+        alice.setPassword("456789");
+        alice.setAge(27);
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void registration_userWithValidData_Ok() {
+        assertEquals(bob, registrationService.register(bob));
+    }
+
+    @Test
+    void addUser_userWithValidData_Ok() {
+        assertEquals(alice, storageDao.add(alice));
```

Comment:
you don't need to test` storageDao.add(alice)` here. Unit test should be separate, and relate to one class, this class only for RegistrationServiceImpl